openedx_events/event_bus/avro/schema.py

```diff
@@ -83,6 +83,21 @@ def _create_avro_field_definition(data_key, data_type, previously_seen_types,
                 f" {set(SIMPLE_PYTHON_TYPE_TO_AVRO_MAPPING.keys())}"
             )
         field["type"] = {"type": PYTHON_TYPE_TO_AVRO_MAPPING[data_type_origin], "items": avro_type}
+    elif data_type_origin == dict:
+        # returns types of dict contents
+        # if data_type == Dict[str, int], arg_data_type = (str, int)
+        arg_data_type = get_args(data_type)
+        if not arg_data_type:
+            raise TypeError(
```

Comment:
Can you add a failure test case?

Reply:
That is supposed to be tested by https://github.com/openedx/openedx-events/blob/acdcaa9c3c73e507c7ca708b94ada2728d463252/openedx_events/event_bus/avro/tests/test_schema.py#L247-L257, I'll check why it's not

Reply:
This one is done! Thanks for pointing it out :)